src/main/java/MyCallableES.java

```diff
@@ -0,0 +1,51 @@
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+import org.apache.commons.collections4.ListUtils;
+
+public class MyCallableES {
+    private static final int THRESHOLD = 20;
+    private List<Integer> list;
+
+    public MyCallableES(List<Integer> list) {
+        this.list = list;
+    }
+
+    public Integer calculateSum() {
+        List<List<Integer>> lists = ListUtils.partition(list, list.size() / THRESHOLD);
+        ExecutorService executorService = Executors.newFixedThreadPool(THRESHOLD);
+
+        List<MyCallable> callableList = lists.stream()
+                .map(MyCallable::new)
+                .collect(Collectors.toList());
+        try {
+            executorService.invokeAll(callableList);
```

Comment:
```suggestion
          List<Future<Integer>> results =  executorService.invokeAll(callableList);
```
then get sums from future